fix: deriving serde on more public structs

benesjan opened the pull request:
In a PR down the stack I needed to slap `#[derive(Deserialize)]` on `RetrievedNote` because I returned it from a function. Nico proposed that we derive serde on more public structs because external won't be able to do so.

In this PR I went through the public structs in Aztec.nr and applied serde where I thought it could make sense to return from a contract function - in most of the cases it didn't make sense.

benesjan commented:
* **#17380** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17380?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17380?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#17325** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17325?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17295** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17295?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17303** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17303?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17283** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/17283?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

benesjan commented on noir-projects/aztec-nr/aztec/src/utils/secrets.nr:
```diff
@@ -1,14 +0,0 @@
-use crate::{hash::compute_secret_hash, oracle::random::random};
-
-pub struct SecretAndHash {
```
This was unused

benesjan commented on noir-projects/noir-protocol-circuits/crates/types/src/abis/nullifier_leaf_preimage.nr:
```diff
@@ -78,28 +77,4 @@ impl NullifierLeafPreimage {
-}
-
-#[test]
-fn serialization_of_empty() {
```
Dropped this test as now, that serde is derived, it seems out of place.

benesjan commented on noir-projects/aztec-nr/compressed-string/src/compressed_string.nr:
```diff
@@ -57,17 +57,6 @@ impl<let N: u32, let M: u32> CompressedString<N, M> {
 
-// TODO: Replace this with the derived version. Not done yet as this uncovered a bug in the AVM. See the discussion
-// below for more details:
-// https://aztecprotocol.slack.com/archives/C02M7VC7TN0/p1754463522334449
```
CI passed so this is no longer an issue. What a nice surprise.

nventuro reviewed (approved):
Thanks! How did you find these and check there are no remnants?

benesjan commented:
>Thanks! How did you find these and check there are no remnants?

@nventuro I just looked through the occurrences of "pub struct" string in noir-projects/aztec-nr and then guessed if it makes sense to ever return from a function.

I didn't apply it to a lot of those structs as I couldn't foresee those to ever be returned (e.g. state variables).

benesjan committed:
fix: deriving serde on more public structs

In a PR down the stack I needed to slap `#[derive(Deserialize)]` on `RetrievedNote` because I returned it from a function. Nico proposed that we derive serde on more public structs because external won't be able to do so.

In this PR I went through the public structs in Aztec.nr and applied serde where I thought it could make sense to return from a contract function - in most of the cases it didn't make sense.
